x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,295 @@
+}
+
+const getInputVarsFields = (
+  input: NewPackagePolicyInput,
+  fields: GcpInputFields[keyof GcpInputFields]
+) =>
+  Object.entries(input.streams[0].vars || {})
+    .filter(([id]) => id in fields)
+    .map(([id, inputVar]) => {
+      const field = fields[id];
+      return {
+        id,
+        label: field.label,
+        type: field.type || 'text',
+        value: inputVar.value,
+      } as const;
+    });
+
+export const GcpCredentialsForm = ({
+  input,
+  newPolicy,
+  updatePolicy,
+  packageInfo,
+  setIsValid,
+  onChange,
+}: Props) => {
+  const fields = getInputVarsFields(input, gcpField.fields);
+
+  useEffect(() => {
+    const isInvalid = packageInfo.version < MIN_VERSION_GCP_CIS;
+
+    setIsValid(!isInvalid);
+
+    onChange({
+      isValid: !isInvalid,
+      updatedPolicy: newPolicy,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [input, packageInfo]);
+
+  if (packageInfo.version < MIN_VERSION_GCP_CIS) {
+    return (
+      <>
+        <EuiSpacer size="l" />
+        <EuiCallOut color="warning">
+          <FormattedMessage
+            id="xpack.csp.gcpIntegration.gcpNotSupportedMessage"
+            defaultMessage="CIS GCP is not supported on the current Integration version, please upgrade your integration to the latest version to use CIS GCP"
+          />
+        </EuiCallOut>
+      </>
+    );
+  }
+  return (
+    <>
+      <GCPSetupInfoContent />
+      <EuiSpacer size="l" />
+      <GcpSetupAccessSelector
+        onChange={(optionId) => updatePolicy(getPosturePolicy(newPolicy, input.type))}
+      />
+      <EuiSpacer size="l" />
+      <GcpInputVarFields
+        fields={fields}
+        onChange={(key, value) =>
+          updatePolicy(getPosturePolicy(newPolicy, input.type, { [key]: { value } }))
+        }
+      />
+      <EuiSpacer size="s" />
+      {DocsLink}
+      <EuiSpacer />
+    </>
+  );
+};
+
+const GcpSetupAccessSelector = ({ onChange }: { onChange(type: GcpCredentialsType): void }) => (
+  <RadioGroup
+    size="s"
+    options={getSetupFormatOptions()}
+    idSelected={'manual'}
+    onChange={(id: GcpCredentialsType) => onChange(id)}
+  />
+);
+
+const GcpInputVarFields = ({
+  fields,
+  onChange,
+}: {
+  fields: Array<GcpFields[keyof GcpFields] & { value: string; id: string }>;
+  onChange: (key: string, value: string) => void;
+}) => {
+  const [credentialOption, setCredentialOption] = useState('Credentials File');
+  const targetFieldName = (id: string) => {
+    return fields.find((element) => element.id === id);
+  };
+  return (
+    <div>
+      <EuiForm component="form">
+        <EuiFormRow fullWidth label={gcpField.fields.project_id.label}>
+          <EuiFieldText
+            id={targetFieldName('project_id')!.id}
+            fullWidth
+            value={targetFieldName('project_id')!.value || ''}
+            onChange={(event) => onChange(targetFieldName('project_id')!.id, event.target.value)}
+          />
+        </EuiFormRow>
```

Comment:
declare a const with the field instead of calling the function multiple times:

```js
const projectIdField = targetFieldName('project_id');
```


also, using the `!` to bypass the non-null checking might lead to errors in this case prefer to render the section only if the field is present:

```jsx
        {projectIdField && (
          <EuiFormRow fullWidth label={gcpField.fields.project_id.label}>
            <EuiFieldText
              id={projectIdField.id}
              fullWidth
              value={projectIdField.value || ''}
              onChange={(event) => onChange(projectIdField.id, event.target.value)}
            />
          </EuiFormRow>
        )}
```

you can apply the same to the other fields